xdsl/transforms/convert_ptr_to_riscv.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from typing import cast
+
+from xdsl.backend.riscv.lowering.utils import (
+    cast_operands_to_regs,
+    register_type_for_type,
+)
+from xdsl.context import MLContext
+from xdsl.dialects import arith, ptr, riscv
+from xdsl.dialects.builtin import (
+    AnyFloat,
+    Float32Type,
+    Float64Type,
+    IndexType,
+    ModuleOp,
+    UnrealizedConversionCastOp,
+)
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    TypeConversionPattern,
+    attr_type_rewrite_pattern,
+    op_type_rewrite_pattern,
+)
+
+
+class PtrTypeConversion(TypeConversionPattern):
+    @attr_type_rewrite_pattern
+    def convert_type(self, typ: ptr.PtrType) -> IndexType:
+        return IndexType()
+
+
+@dataclass
+class ConvertPtrAddOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.PtrAddOp, rewriter: PatternRewriter, /):
+        rewriter.replace_matched_op(arith.Addi(op.operands[0], op.operands[1]))
+
+
+@dataclass
+class ConvertStoreOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.StoreOp, rewriter: PatternRewriter, /):
+        addr, value = cast_operands_to_regs(rewriter)
+
+        match value.type:
+            case riscv.IntRegisterType():
+                new_op = riscv.SwOp(
+                    addr, value, 0, comment="store int value to pointer"
+                )
+            case riscv.FloatRegisterType():
+                float_type = cast(AnyFloat, op.value.type)
+                match float_type:
+                    case Float32Type():
+                        new_op = riscv.FSwOp(
+                            addr,
+                            value,
+                            0,
+                            comment="store float value to pointer",
+                        )
+                    case Float64Type():
+                        new_op = riscv.FSdOp(
+                            addr,
+                            value,
+                            0,
+                            comment="store double value to pointer",
+                        )
+                    case _:
+                        assert False, f"Unexpected floating point type {float_type}"
```

Comment:
```suggestion
                        raise DiagnosticException(f"Unexpected floating point type {float_type}")
```
This should raise a diagnostic exception, and we should add a filecheck test to check that we print a helpful string.